src/main/kotlin/world/trait/Trait.kt

```diff
@@ -70,9 +88,47 @@ open class Trait(val name: String) {
     open fun callVerb(c: Context, verbName: String, args: List<Value>): Value? {
         verbs[verbName]?.also {
             Log.i("static verb call: \$$name.$verbName($args)")
-            return it.call(c, Yegg.vNullObj, vTrait, args)
+            return it.call(c, c.vThis, vTrait, args)
         }
         return null
     }
 
+    fun matchCommand(obj: Obj?, cmdstr: String, argstr: String, dobjstr: String, dobj: Obj?, prep: Preposition?, iobjstr: String, iobj: Obj?): CommandMatch? {
+        fun matchArg(t: Obj?, arg: Arg?, s: String, o: Obj?): Value? =
```

Comment:
It looks like `t` will always be `obj`, so I think you don't really need this param.  You'll get `obj` from the enclosing scope, right?  I am not 100% sure how nested functions work in kotlin anymore, so plz double check my assumptions.

Reply:
Nitpick: it would be nice if the names of the params were more descriptive instead of just single letters.  It took me a while to make sure I was reading things correctly, and that would have been a lot easier if they had names like `thisObj`, `matchObj`, etc.

Reply:
> It looks like `t` will always be `obj`, so I think you don't really need this param. You'll get `obj` from the enclosing scope, right? I am not 100% sure how nested functions work in kotlin anymore, so plz double check my assumptions.

Nice catch, changed ⛳

Reply:
> Nitpick: it would be nice if the names of the params were more descriptive instead of just single letters. It took me a while to make sure I was reading things correctly, and that would have been a lot easier if they had names like `thisObj`, `matchObj`, etc.

Thanks, I have an irrational tendency to make code in inner functions more cryptic, like it needs to fit inside or something.  Changed to be readable!